### DISCUSSION TITLE
pac solution pack should support plugin packages

### DISCUSSION THREAD

**Discussion by mathiasbl** [microsoft/powerplatform-build-tools#670] [Ideas]
Been trying to replace our own built nuget package by using the mapping file functionality during the pac solution pack command but it's not getting picked up.

Is this feature supported or still in the works.

**Comment by mathiasbl** [comment]
@JimDaly sorry, actually meant the pac solution pack command.

**Reply by JimDaly** [reply]
OK. I deleted my comment.

**Comment by jworbis** [comment]
Hi,
I was about to ask the same. I was trying to map my pluginpackages but couldn't get it to work. Is there anything planned for the future? Or is there a workaround? I didn't have the time to test it completely, I somehow managed to create a map xml that deletes the nuget package on pac solution unpack, but I had no luck mapping my build outputs into the solution.

Thanks!

**Comment by mathiasbl** [comment]
@jworbis They added an example in the docs.

```
<FileToFile
    map="pluginpackages\cr886_PluginPackageTest\package\cr886_PluginPackageTest.nupkg"
    to="myplg\bin\Debug\myplg.1.0.0.nupkg" />
```

https://learn.microsoft.com/en-us/power-platform/alm/solution-packager-tool

**Comment by ferodom** [comment]
I am having an issue when unpacking plugin nuget packages, if there is a map to a file that is not part of the solution I get the error below. Note Having multiple plugin directives work just fine, the tool just skips them. Note: I prefer to have just one map.xml since I work with multiple solutions.

Error: Value cannot be null.
Parameter name: key

this is my map.xml

<?xml version="1.0" encoding="utf-8"?>
<Mapping>
    
  <FileToFile map="BswCrmPlugins.dll" to="..\..\..\Bsw.Crm.Plugins\bin\**\Bsw.Crm.Plugins.dll" />
  <FileToFile map="BswCrmMAPlugins.dll" to="..\..\..\Bsw.Crm.MA.Plugins\bin\**\Bsw.Crm.MA.Plugins.dll" />
  <FileToFile map="BswCrmMDMPlugins.dll" to="..\..\..\Bsw.Crm.MDM.Plugins\bin\**\Bsw.Crm.MDM.Plugins.dll" />
  <FileToFile map="BswCrmPersonalizationPlugins.dll" to="..\..\..\Bsw.Crm.Personalization.Plugins\bin\**\Bsw.Crm.Personalization.Plugins.dll" />  
  <FileToFile map="pluginpackages\bsw_Bsw.Crm.CS.CustomApis\package\bsw_Bsw.Crm.CS.CustomApis.nupkg" to="..\..\..\Bsw.Crm.CS.CustomApis\bin\**\Bsw.Crm.CS.CustomApis.1.0.0.nupkg" />
  <FileToFile map="pluginpackages\bsw_Bsw.Crm.FCN.CustomApis\package\bsw_Bsw.Crm.FCN.CustomApis.nupkg" to="..\..\..\Bsw.Crm.FCN.CustomApis\bin\**\Bsw.Crm.FCN.CustomApis.1.0.0.nupkg" />
  <FileToFile map="WebResources\bsw_\providersearch\index.html" to="..\..\..\Bsw.Crm.ProviderSearch\dist\index.html" />
  <FileToFile map="WebResources\bsw_\providersearch\main.js" to="..\..\..\Bsw.Crm.ProviderSearch\dist\main.js" />  
  
   
  <FileToPath map="WebResources\bsw_\styles\*.*" to="..\..\..\Bsw.Crm.WebResources\**" />
  <FileToPath map="WebResources\bsw_\views\*.*" to="..\..\..\Bsw.Crm.WebResources\**" />
  <FileToPath map="WebResources\bsw_\scripts\*.*" to="..\..\..\Bsw.Crm.WebResources\**" />

  
  <FileToPath map="WebResources\bsw_\inhomesupporttool\*.*" to="..\..\..\Bsw.Crm.InHomeSupportTool\dist\**" />
  
</Mapping>

**Reply by SakthiChandra** [reply]
I'm encountering the same error when unpacking the solution, particularly when plugin packages are included in my filemap XML. How can I resolve this issue?

**Reply by janis-veinbergs** [reply]
I founda workaround - move `FileToFile` plugin package BEFORE all mappings. In my case it was after some other plugins and I got this error. Once moved to top, `unpack` works. Screams like a bug to me.

**Reply by ferodom** [reply]
I tried that as well and it worked for me as long as I had one plugin package, Once I added the other then it broke again. Definitely a bug. Can we get some help?
